sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -432,10 +432,10 @@ private bool TryGetReturnTypeBindings(MethodDeclarationSyntax method, SemanticMo
                 }
 
                 if (!SymbolEqualityComparer.Default.Equals(returnTypeSymbol, _knownTypes.VoidType) &&
-                    !SymbolEqualityComparer.Default.Equals(returnTypeSymbol, _knownTypes.TaskType))
+                    !SymbolEqualityComparer.Default.Equals(returnTypeSymbol.OriginalDefinition, _knownTypes.TaskType))
```

Comment:
For an async function, `returnTypeSymbol` here is `System.Threading.Tasks.Task<Microsoft.Azure.Functions.Worker.Http.HttpResponseData>` and `.OriginalDefinition` is `System.Threading.Tasks.Task<TResult>` - which is same as what we are comparing against here (`_knownTypes.TaskType`)